src/DotVVM.Samples.Common/Views/ControlSamples/TemplateHost/CompositeControlWithTemplate.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Text;
+using DotVVM.Framework.Binding;
+using DotVVM.Framework.Controls;
+
+namespace DotVVM.Samples.Common.Views.ControlSamples.TemplateHost
+{
+    public class CompositeControlWithTemplate : CompositeControl 
+    {
+
+        public static DotvvmControl GetContents(
+            ValueOrBinding<string> headerText,
+            ITemplate contentTemplate
+        )
+        {
+            return new HtmlGenericControl("fieldset")
+                .AppendChildren(
+                    new HtmlGenericControl("legend", new TextOrContentCapability() { Text = headerText }),
+                    new Framework.Controls.TemplateHost() { Template = contentTemplate }
```

Comment:
Is there some naming conflict?

```suggestion
                    new TemplateHost() { Template = contentTemplate }
```

Reply:
Yes, it's correct. 
The folders in control samples have the same names as the controls being tested, so the class is in the `TemplateHost` namespace. :-(